python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -1050,19 +1050,18 @@ def reset_index(self, drop=False):
         return out.set_index(RangeIndex(len(self)))
 
     def take(self, positions, ignore_index=False):
-        positions = columnops.as_column(positions).astype("int32").data.mem
         out = DataFrame()
-        cols = [s._column for s in self._cols.values()]
-
-        result_cols = cpp_copying.apply_gather(cols, positions)
-
-        for i, col_name in enumerate(self._cols):
-            out[col_name] = result_cols[i]
-
-        if ignore_index:
-            out.index = RangeIndex(len(out))
-        else:
-            out.index = self.index.take(positions)
+        if self._cols:
```

Comment:
It shouldn't be possible to have a DataFrame that doesn't have a `self._cols`, did you run into an issue with this clause in testing?

Reply:
I used that to determine if there are any columns in the DataFrame, since 0 columns causes an `IndexError` from `cpp_copying.apply_gather()` *(because it tries to check the `dtype` of the first item in the column list<sup>1</sup>)*.  This results in identical behavior to the now-removed `_sort_by()`, while (I think) preserving the intent of `take()` on a DataFrame with no columns. I discovered that in testing and ensured the tests passed after adding that check.

Do you think it would be clearer if it was `if len(self._cols) > 0:` ?

<sup>1</sup> *should having `cpp_copying.apply_gather()` handle an empty column list be a separate issue?*

Reply:
That answers my question - `self._cols:` sounds good here.